.github/bin/filter-commits-to-compile.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import logging
+import sys
+import tempfile
+import unittest
+
+import csv
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Choose commits to compile using their CSV descriptions."
+    )
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=argparse.FileType("r"),
+        default=sys.stdin,
+        help="A CSV file in <commit hash>,<tree hash>,<commit subject>",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Filename to write chosen commit hashes to",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_const",
+        dest="loglevel",
+        const=logging.INFO,
+        default=logging.WARNING,
+        help="Print info level logs",
+    )
+    parser.add_argument(
+        "-t",
+        "--test",
+        action='store_true',
+        help="test this script instead of executing it",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.loglevel)
+    if args.test:
+        sys.argv = [sys.argv[0]]
+        unittest.main()
+        return
+    build(args.input, args.output)
+
+
+def build(input_file, output_file):
+    logging.info("input_file: %s", input_file)
+    logging.info("output_file: %s", output_file)
+    reader = csv.reader(input_file)
+    entries = list(reader)
+    logging.info("entries: %s", entries)
+    for i, entry in enumerate(entries):
+        commit_hash = entry[0]
+        # TODO: add filtering based on GitHub Actions cache entries
```

Comment:
Should we have an issue to track this? I don't fully understand why we'd want to do this.

Reply:
It's an optimization for later so that we don't have to wait for a GH runner and download the entire repo only to find that this tree hash was already compiled successfully.